themes/university.typ

```diff
@@ -0,0 +1,263 @@
+                rows: (1mm, auto),
+                block(fill: color-b, width:100%)[
+                    #locate(loc => {
+                        let ratio = logical-slide.at(loc).first() / logical-slide.final(loc).first() * 100%
+                        grid(
+                            columns: (ratio, auto),
+                            cell(fill: color-a, height: 1mm)[],
+                        )
+                    }
+                    )
+                ],
+                block(fill: color-c.lighten(90%))[
+                    #grid(
+                        columns: (60%, 40%),
+                        cell[
+                            #if "title" in slide-info {
+                              heading(level: 2, text(color-title)[#slide-info.title])
+                            }
+                        ],
+                        cell[
+                            #align(right)[
+                                #heading(level: 2, text(color-section)[#section.display()])
+                            ]
+                        ]
+                    )
+                ],
+            )
+        }
+
+        let footer-deco() = {
+            let cell = rect.with(
+                width: 100%,
+                inset: 1mm,
+                outset: 0mm
+            )
+            set text(
+                size: 10pt
+            )
+            set align(center)
+            grid(
+                columns: (25%, 1fr, 15%, 10%),
+                rows: (5mm, auto),
+                cell(
+                    height: 100%,
+                    fill: color-footer-fill-short-authors
+                )[#text(color-footer-short-authors)[#data.short-authors]],
+                cell(
+                    height: 100%,
+                    fill: color-footer-fill-short-title
+                )[#text(color-footer-short-title)[#data.short-title]],
+                cell(
+                    height: 100%,
+                    fill: color-footer-fill-date
+                )[#text(color-footer-date)[#data.date]],
+                cell(
+                    height: 100%,
+                    fill: color-footer-fill-slide-counter
+                )[#text(color-footer-slide-counter)[#logical-slide.display() / #last-slide-number]]
+            )
+        }
+
+        // header
+        header-deco()
+
+        v(1fr)
+        block(
+            width: 100%, inset: (x: 1em), breakable: false, outset: 0em,
+            body
+        )
+        v(2fr)
+
+        // footer
+        footer-deco()
+    }
+
+    let wake-up(slide-info, bodies) = {
+        if bodies.len() != 1 {
+            panic("wake-up variant of university theme only supports one body per slide")
+        }
+
+        let body = bodies.first()
+
+        block(
+            width: 100%,
+            height: 100%,
+            inset: 2em,
+            breakable: false,
+            outset: 0em,
+            fill: if "color" in slide-info {slide-info.color} else {black},
```

Comment:
How about special-casing here for the strings `"a"` to `"d"`? Then users can easily make wake up slides consistent with the rest of the team.

Reply:
Can you please elaborate?

Reply:
See my earlier comment:
https://github.com/andreasKroepelin/typst-slides/pull/21#discussion_r1170000278

Reply:
But the `wake up`  variant is able to accept a color argument. Is it still relevant then?